### ISSUE TITLE
SoundEffect issues (Windows DX, master branch)

### ISSUE THREAD

**Issue by aienabled** [MonoGame/MonoGame#5466]
Hi guys,

I just started working with audio system for my game. I don't use MonoGame pipeline and load all the assets in runtime from their source files. So I'm using these constructors for `SoundEffect` (accepting PCM wave data):
```C#
SoundEffect(byte[] buffer, int sampleRate, AudioChannels channels)
SoundEffect(byte[] buffer, int offset, int count, int sampleRate, AudioChannels channels, int loopStart, int loopLength)
```

However, I've found two issues with the first method (the second method works fine):
1. The sound played much longer than specified, and during the added duration it produces various noise.
2. With some specific effect duration I have a native crash when try to play the sound.

I've created a sample game reproducing the issue. It's very simple - it creates the sine wave (as float[] samples array) and then convert it to the byte array (PCM wave format).
On mouse left button click it plays the sound and writes to the output log (VS output log) how long it was playing: `Sound played duration: 1366.7ms - vs - sound effect duration 680.0ms` (in case of the first method, in case of the second method it's exactly as it should be (if the sample count matches frame rate).
Also there is a line to uncomment `//var samplesCount = 44100;` to test the native crash issue.

You can get the sample game code there https://gist.github.com/aienabled/e0589a23edcb933fba2277c5152041e9 (just create a new MonoGame DX project and replace `Game1.cs` with this file).

Regards!

**Comment by mrhelmut** [comment]
Hi,

Would you mind switching to the develop branch? I believe a bunch of improvements have been made in that regard.

**Comment by aienabled** [comment]
Sorry I was not able to update to the develop branch.
I can confirm that the issue is resolved in master branch (v3.6).
Regards!
